### PR TITLE
[2.13.x] DDF-4049 Fixes DefaultErrorHandler so that it generates correct, safe response

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### What does this PR do?
The old implementation leaked internals and required Javascript for dynamic page rendering
whereas the new implementation generates the page statically on the backend.
This also removes the defunct ui module that had been associated with the error handler.

**N.B. This PR is a duplicate of the merged, then reverted #3578. That PR was reverted along with #3585 to facilitate the release of 2.13.1. There appears to be an issue with bundle restarting in that other PR that needs to be addressed. Review of this PR should be minimal as it is identical to the version already approved.**

#### Who is reviewing it? 

@Kjames5269 @rzwiefel  

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@djblue 
@stustison

#### How should this be tested?

Any unhandled exception that can be generated from the backend should trigger this. During development, I modified `WebSSOFilter` and `IdpEndpoint` to force them to throw `RuntimeException`, but any unhandled exception that can be generated on a web call should suffice.

The page should display and look something like the attached image below, with the specifics of error code and the two messages distinct for a given response.

#### Any background context you want to provide?
*N.B. This work cannot be tested or merged until the changes for [DDF-4033](https://codice.atlassian.net/browse/DDF-4033) are completed. A PR for that should be forthcoming in the next few days.*

#### What are the relevant tickets?
[DDF-4049](https://codice.atlassian.net/browse/DDF-4049)
[DDF-4033](https://codice.atlassian.net/browse/DDF-4033)
#### Screenshots

screen shot 2018-08-09 at 11 57 15 am

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/error/platform-error-impl/src/main/java/org/codice/ddf/platform/error/handler/impl/DefaultErrorHandler.java [hunk 1]
```diff
@@ -13,17 +13,16 @@
  */
 package org.codice.ddf.platform.error.handler.impl;
 
-import static org.boon.Boon.toJson;
```
Comment (rzwiefel) [inline platform/error/platform-error-impl/src/main/java/org/codice/ddf/platform/error/handler/impl/DefaultErrorHandler.java]: ❓ If we removed boon does that mean we can remove the dependency from the pom or do other things in here still use it?

Comment (coyotesqrl) [reply platform/error/platform-error-impl/src/main/java/org/codice/ddf/platform/error/handler/impl/DefaultErrorHandler.java]: Good catch. I should be able to remove that dependency.

Comment (coyotesqrl) [reply platform/error/platform-error-impl/src/main/java/org/codice/ddf/platform/error/handler/impl/DefaultErrorHandler.java]: Removed boon dependency.

**Comment by bakejeyner** [comment]
Hero ✔️

**Comment by coyotesqrl** [comment]
build now

**Pull request merged**